### PR TITLE
fix(antiddos/precheck): update refresh logic of the precheck

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The Anti-DDoS basic resource need a few time before creation and after EIP creation complete to make sure the service configuration synchronization complete.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. the pre-check need be supported before Anti-DDoS creation if the region is not 'cn-north-4'.
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

**cn-north-4**
```
make testacc TEST='./huaweicloud/services/acceptance/antiddos' TESTARGS='-run=TestAccAntiDdos_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/antiddos -v -run=TestAccAntiDdos_basic -timeout 360m -parallel 4
=== RUN   TestAccAntiDdos_basic
=== PAUSE TestAccAntiDdos_basic
=== CONT  TestAccAntiDdos_basic
--- PASS: TestAccAntiDdos_basic (83.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/antiddos  83.168s
```

**af-south-1**
```
make testacc TEST='./huaweicloud/services/acceptance/antiddos' TESTARGS='-run=TestAccAntiDdos_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/antiddos -v -run=TestAccAntiDdos_basic -timeout 360m -parallel 4
=== RUN   TestAccAntiDdos_basic
=== PAUSE TestAccAntiDdos_basic
=== CONT  TestAccAntiDdos_basic
--- PASS: TestAccAntiDdos_basic (273.55s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/antiddos  273.591s
```

**cn-south-1**
```
make testacc TEST='./huaweicloud/services/acceptance/antiddos' TESTARGS='-run=TestAccAntiDdos_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/antiddos -v -run=TestAccAntiDdos_basic -timeout 360m -parallel 4
=== RUN   TestAccAntiDdos_basic
=== PAUSE TestAccAntiDdos_basic
=== CONT  TestAccAntiDdos_basic
--- PASS: TestAccAntiDdos_basic (114.44s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/antiddos  114.478s
```

### CODE REVIEW

File: huaweicloud/services/antiddos/resource_huaweicloud_antiddos_basic.go [hunk 1]
```diff
@@ -245,3 +278,22 @@ func getAntiDdosStatus(client *golangsdk.ServiceClient, antiddosID string) resou
...
+	ErrCode string `json:"error_code"`
+	// Error message
+	ErrMessage string `json:"error_msg"`
+}
+
+func parseAntiDDoSQueryError(respErr error) error {
+	var apiError antiDDoSNotFoundError
+
+	if errCode, ok := respErr.(golangsdk.ErrDefault403); ok {
+		pErr := json.Unmarshal(errCode.Body, &apiError)
```
Comment (ShiChangkuo) [inline huaweicloud/services/antiddos/resource_huaweicloud_antiddos_basic.go]: you can try to use `common.ParseErrorMsg(errCode.Body)`

File: huaweicloud/services/antiddos/resource_huaweicloud_antiddos_basic.go [hunk 2]
```diff
@@ -97,6 +98,43 @@ func ResourceCloudNativeAntiDdos() *schema.Resource {
...
+	timeout time.Duration, isDeleteCheck bool) (*antiddossdk.GetResponse, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"PENDING"},
+		Target:  []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			resp, err := antiddossdk.Get(client, antiDDoSId).Extract()
+			if err != nil {
+				analyzedErr := parseAntiDDoSQueryError(err)
+				_, ok := analyzedErr.(golangsdk.ErrDefault404)
+				if !ok {
```
Comment (ShiChangkuo) [inline huaweicloud/services/antiddos/resource_huaweicloud_antiddos_basic.go]:
```suggestion
				if _, ok := analyzedErr.(golangsdk.ErrDefault404); !ok {
```

File: huaweicloud/services/antiddos/resource_huaweicloud_antiddos_basic.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/antiddos/resource_huaweicloud_antiddos_basic.go]: okey, it works

File: huaweicloud/services/antiddos/resource_huaweicloud_antiddos_basic.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/antiddos/resource_huaweicloud_antiddos_basic.go]: done.

**Pull request merged**